app/docs/md/learn/concepts/routing/dynamic-routes.md

```diff
@@ -0,0 +1,44 @@
+---
+title: Dynamic Routes
+---
+
+Predefined paths are not always enough for complex applications. In Enhance you can add `$` to a `page` or `api` filename to create a dynamic route.
+
+## Example
+
+Consider the following api `app/api/users/$id.mjs`:
+
+<doc-code filename="app/api/users/$id.mjs">
+
+```javascript
+export async function get (req) {
+  const { id } = req.pathParameters
+  const user = getUser(id)
+  return {
+    json: {
+     user
+    }
+  }
+}
+```
+</doc-code>
+
+Routes like `/users/axol` or `/users/admin` will be matched by `app/api/users/$id.mjs`. The `user` JSON object will be returned to the caller if the request's headers include `Accept: application/json` otherwise the route will call `app/pages/users/$id.mjs` if the file exists.
```

Comment:
Noob question: if the route ends up calling `app/pages/users/$id.mjs`, will the `id` parameter be exposed to SFCs on that page in some way automatically? Or would I still need to author an API route to manage that?

Reply:
You would need an API to put `req.params` into the store.

Reply:
Makes sense, thanks!